File: examples/java/basic.java
import com.apiverve.invoicegenerator.InvoiceGeneratorAPIClient;
import com.apiverve.invoicegenerator.APIResponse;
import com.apiverve.invoicegenerator.APIException;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class BasicExample {
    public static void main(String[] args) {
        // Initialize the API client with your API key
        InvoiceGeneratorAPIClient client = new InvoiceGeneratorAPIClient("YOUR_API_KEY_HERE");

        try {
            // Request body
            Map&lt;String, Object&gt; parameters &#x3D; new HashMap&lt;&gt;();
        parameters.put(&quot;invoiceNumber&quot;, &quot;INV000001&quot;);
        parameters.put(&quot;date&quot;, &quot;2025-02-01&quot;);
        parameters.put(&quot;dueDate&quot;, &quot;2025-11-30&quot;);
        parameters.put(&quot;from_name&quot;, &quot;John Doe&quot;);
        parameters.put(&quot;from_street&quot;, &quot;123 Elm St&quot;);
        parameters.put(&quot;from_city&quot;, &quot;Springfield&quot;);
        parameters.put(&quot;from_state&quot;, &quot;IL&quot;);
        parameters.put(&quot;from_zip&quot;, &quot;62701&quot;);
        parameters.put(&quot;to_name&quot;, &quot;Jane Smith&quot;);
        parameters.put(&quot;to_street&quot;, &quot;456 Oak St&quot;);
        parameters.put(&quot;to_city&quot;, &quot;Springfield&quot;);
        parameters.put(&quot;to_state&quot;, &quot;IL&quot;);
        parameters.put(&quot;to_zip&quot;, &quot;62702&quot;);
        parameters.put(&quot;job&quot;, &quot;Web Development&quot;);
        parameters.put(&quot;paymentTerms&quot;, &quot;Net 30&quot;);
        parameters.put(&quot;discount&quot;, 10);
        parameters.put(&quot;salesTax&quot;, 37.07);
        parameters.put(&quot;currency&quot;, &quot;USD&quot;);
        parameters.put(&quot;items&quot;, [object Object],[object Object]);

            // Execute the API request
            APIResponse response = client.execute(parameters);

            // Check if the request was successful
            if (response.isSuccess()) {
                System.out.println("✅ Success!");

                // Get the response data
                JSONObject data = response.getData();
                if (data != null) {
                    System.out.println("Response data:");
                    System.out.println(data.toString(2)); // Pretty print with 2-space indent
                }

                // Or get specific fields from the data
                // String value = data.optString("fieldName");

            } else {
                // Handle API errors
                System.err.println("❌ API Error: " + response.getError());
                System.err.println("Status: " + response.getStatus());
                System.err.println("Code: " + response.getCode());
            }

        } catch (APIException e) {
            // Handle exceptions
            System.err.println("❌ Error: " + e.getMessage());

            // Handle specific error types
            if (e.isAuthenticationError()) {
                System.err.println("Invalid API key. Get your key at: https://apiverve.com");
            } else if (e.isRateLimitError()) {
                System.err.println("Rate limit exceeded. Please try again later.");
            } else if (e.isServerError()) {
                System.err.println("Server error (5xx). Please try again later.");
            } else if (e.isClientError()) {
                System.err.println("Client error (4xx). Check your request parameters.");
            }

            // Get HTTP status code if available
            if (e.getStatusCode() > 0) {
                System.err.println("HTTP Status Code: " + e.getStatusCode());
            }
        }
    }
}

File: examples/csharp/basic.cs
/*
 * Invoice Generator API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Invoice Generator API.
 * API Documentation: https://docs.apiverve.com/ref/invoicegenerator
 */

using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using System.Linq;

namespace APIVerve.Examples
{
    class InvoiceGeneratorExample
    {
        private static readonly string API_KEY = Environment.GetEnvironmentVariable("APIVERVE_API_KEY") ?? "YOUR_API_KEY_HERE";
        private static readonly string API_URL = "https://api.apiverve.com/v1/invoicegenerator";

        /// <summary>
        /// Make a POST request to the Invoice Generator API
        /// </summary>
        static async Task<JsonDocument> CallInvoiceGeneratorAPI()
        {
            try
            {
                using var client = new HttpClient();
                client.DefaultRequestHeaders.Add("x-api-key", API_KEY);

                // Request body
                var requestBody &#x3D; new { invoiceNumber &#x3D; &quot;INV000001&quot;, date &#x3D; &quot;2025-02-01&quot;, dueDate &#x3D; &quot;2025-11-30&quot;, from_name &#x3D; &quot;John Doe&quot;, from_street &#x3D; &quot;123 Elm St&quot;, from_city &#x3D; &quot;Springfield&quot;, from_state &#x3D; &quot;IL&quot;, from_zip &#x3D; &quot;62701&quot;, to_name &#x3D; &quot;Jane Smith&quot;, to_street &#x3D; &quot;456 Oak St&quot;, to_city &#x3D; &quot;Springfield&quot;, to_state &#x3D; &quot;IL&quot;, to_zip &#x3D; &quot;62702&quot;, job &#x3D; &quot;Web Development&quot;, paymentTerms &#x3D; &quot;Net 30&quot;, discount &#x3D; 10, salesTax &#x3D; 37.07, currency &#x3D; &quot;USD&quot;, items &#x3D; [object Object],[object Object] };

                var jsonContent = JsonSerializer.Serialize(requestBody);
                var content = new StringContent(jsonContent, Encoding.UTF8, "application/json");

                var response = await client.PostAsync(API_URL, content);

                // Check if response is successful
                response.EnsureSuccessStatusCode();

                var responseBody = await response.Content.ReadAsStringAsync();
                var data = JsonDocument.Parse(responseBody);

                // Check API response status
                if (data.RootElement.GetProperty("status").GetString() == "ok")
                {
                    Console.WriteLine("✓ Success!");
                    Console.WriteLine("Response data: " + data.RootElement.GetProperty("data"));
                    return data;
                }
                else
                {
                    var error = data.RootElement.TryGetProperty("error", out var errorProp)
                        ? errorProp.GetString()
                        : "Unknown error";
                    Console.WriteLine($"✗ API Error: {error}");
                    return null;
                }
            }
            catch (HttpRequestException e)
            {
                Console.WriteLine($"✗ Request failed: {e.Message}");
                return null;
            }
        }

        static async Task Main(string[] args)
        {
            Console.WriteLine("📤 Calling Invoice Generator API...\n");

            var result = await CallInvoiceGeneratorAPI();

            if (result != null)
            {
                Console.WriteLine("\n📊 Final Result:");
                Console.WriteLine(result.RootElement.GetProperty("data"));
            }
            else
            {
                Console.WriteLine("\n✗ API call failed");
            }
        }
    }
}
